CTestConfig.cmake

```diff
@@ -34,7 +34,7 @@ else ()
   if (CDASH_LOCAL)
     set (CTEST_DROP_LOCATION "/submit.php?project=HDF5114")
   else ()
-    set (CTEST_DROP_LOCATION "/submit.php?project=HDF5")
+    set (CTEST_DROP_LOCATION "/submit.php?project=HDF5114")
```

Comment:
Looks like these two lines for setting `CTEST_DROP_LOCATION` are exactly the same?

Reply:
Unclear why we do things differently w/ CDASH_LOCAL in the first place